### PR TITLE
cpu, cc2538: adapt periph i2c to GPIO API

### PR DESCRIPTION
**Pull request by smlng**
This PR requires #7316 and 
- adapts low level I2C periph driver of cc2538 to use RIOTs GPIO API
- enhance coding style 
- adapt board periph_conf for cc2538 based boards

### CODE REVIEW

**Comment by smlng** [comment]
@PeterKietzmann and ... well you know.

**Comment by smlng** [comment]
rebased on (and includes) #7373

**Comment by smlng** [comment]
rebased, needs squashing and testing

File: cpu/cc2538/periph/i2c.c [hunk 1]
```diff
@@ -26,33 +27,33 @@
...
+#include "thread.h"
 #include "timex.h" /* for US_PER_SEC */
+#endif
 
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
 /* guard this file in case no I2C device is defined */
-#if I2C_NUMOF
+#ifdef I2C_NUMOF
```
Comment (PeterKietzmann) [inline cpu/cc2538/periph/i2c.c]: This is not needed anymore according to #7981

File: cpu/cc2538/periph/i2c.c [hunk 2]
```diff
@@ -26,33 +27,33 @@
...
 #include "mutex.h"
 #include "cpu.h"
+#include "periph/gpio.h"
 #include "periph/i2c.h"
-#include "thread.h"
 #ifdef MODULE_XTIMER
 #include "xtimer.h"
-#endif
+#else
+#include "thread.h"
```
Comment (PeterKietzmann) [inline cpu/cc2538/periph/i2c.c]: I won't NACK this PR for this change but I don't see what this has to do with the i2c implementation

File: cpu/cc2538/periph/i2c.c [hunk 3]
```diff
@@ -26,33 +27,33 @@
...
-#define I2C_0_SCL_PIN i2c_config[I2C_0].scl_pin
-#endif
+/* pin short cuts */
+#define SCL_PIN(x) (i2c_config[x].scl_pin)
+#define SDA_PIN(x) (i2c_config[x].sda_pin)
 
-#ifndef I2C_0_SDA_PIN
-#define I2C_0_SDA_PIN i2c_config[I2C_0].sda_pin
-#endif
+#define SDA_TRY_LIMIT   (200U)
```
Comment (PeterKietzmann) [inline cpu/cc2538/periph/i2c.c]: 1. Where is this used and 2. could you spend a line explaining what it does?

File: cpu/cc2538/periph/i2c.c [hunk 4]
```diff
@@ -85,79 +86,17 @@
...
-
-        if (n >= try_limit) {
-            DEBUG("%s(): Failed to release SDA after%4u SCL pulses.\n", __FUNCTION__, n);
-        }
-    }
-
-    /* Return to hardware mode for the I2C pins */
-    gpio_hardware_control(I2C_0_SCL_PIN);
-    gpio_hardware_control(I2C_0_SDA_PIN);
+static inline void i2cm_ctrl_write(uint_fast8_t value) {
```
Comment (PeterKietzmann) [inline cpu/cc2538/periph/i2c.c]: Above it was 
```
static void i2cm_ctrl_write(uint_fast8_t value) {
    WARN_IF(I2CM_STAT & BUSY);
    I2CM_CTRL = value;
}
```
Why did you remove the `WARN_IF` logging? Here and following?

**Comment by smlng** [comment]
@PeterKietzmann thx for testing, but that confirms my failed testing during HnA ☹️

**Comment by A-Paul** [comment]
Tested on *cc2538dk* with same outcome as @PeterKietzmann. Also tried more low-level communication with *tests/periph_i2c*. But there's no i2c communication at all, either with 100kbit/s or 10kbit/s.

**Comment by tcschmidt** [comment]
@smlng fix?

**Comment by aabadie** [comment]
@smlng, is this PR still waiting for another PR (it doesn't seem to be the case) ? Do you think you could adapt it to the new I2C rework ?

The idea would be to rebase it on top of new_i2c_if branch and change the base branch (click 'edit' button at the top right of this page and change the base branch there). And then link this PR in #6577. That would be super nice.

**Comment by smlng** [comment]
FYI, @aabadie and all: closing and reopening a new PR against `new_i2c_if` branch later.

**Comment by aabadie** [comment]
Thanks @smlng !
